lua/lint/linters/pylint.lua

```diff
@@ -12,28 +12,31 @@ return {
     '-f', 'json'
   },
   ignore_exitcode = true,
-  parser = function(output)
-    local decoded = vim.fn.json_decode(output)
+  parser = function(output, bufnr)
     local diagnostics = {}
-    for _, item in ipairs(decoded or {}) do
-      local column = 0
-      if item.column > 0 then
-        column = item.column - 1
-      end
-      table.insert(diagnostics, {
-        range = {
-          ['start'] = {
-            line = item.line - 1,
-            character = column,
-          },
-          ['end'] = {
-            line = item.line - 1,
-            character = column,
+    local buffer_path = vim.fn.fnamemodify(vim.api.nvim_buf_get_name(bufnr), ":~:.")
+
+    for _, item in ipairs(vim.fn.json_decode(output) or {}) do
+      if not item.file or vim.fn.fnamemodify(item.file, ":~:.") == buffer_path then
```

Comment:
The test cases use `path` instead of `file`, is that intentional?

Reply:
Didn't notice that. It uses path in the actual output, and since expected behavior is it either matches or is undefined, I wouldn't have caught it.

Fixed in b6bba9f.